### ISSUE TITLE
Save data corresponding to plot

### ISSUE THREAD

**Issue by VanessaBa** [esqLABS/esqlabsR#289]
﻿it would be useful when, everytime a plot is saved, automatically the corresponding data is saved as an xls or csv along with the plot. this would ensure that at least the plot can be reproduced or adapted

**Comment by PavelBal** [comment]
Not relevant

**Comment by VanessaBa** [comment]
not funny!

**Comment by IndrajeetPatil** [comment]
If it's a ggplot2 plot, the data will be stored in the plot object:

``` r
library(ggplot2)

p <- ggplot(mtcars, aes(wt, mpg)) + geom_point() + geom_smooth()

p
```

![](https://i.imgur.com/5BKm3sT.png)

``` r

pb <- ggplot_build(p)

identical(mtcars, pb$plot$data)
#> [1] TRUE
```

<sup>Created on 2022-07-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Not sure about base graphics; I don't use it.

**Comment by VanessaBa** [comment]
I see. but what I would want to have is a excel file, such that when I grab out some plot (also from months ago) I still can access the data on which the plot is based. So automatically writing the the underlying dataframe or datatable to an xls would probably do the job.

**Comment by IndrajeetPatil** [comment]
Yeah, when you create a plot object `p`, you can just immediately add a call to a custom function:

```r
plotDataToCSV <- function(plotObject, filename) {
  stopifnot(inherits(plotObject, "ggplot"))
  pb <- ggplot_build(plotObject)
  readr::write_csv(pb$plot$data, filename)
}

plotDataToCSV(p, "p.csv")
```

**Comment by PavelBal** [comment]
For now, we can implement a function to write a `DataMapping` to excel (resp. something like `dataMappingToDataFrame()`). This should be done in `esqlabsRlegacy`. Vanessa you can just update the legacy package without the risk of your project becoming incompatible with the latest esqlabsR.

**Comment by VanessaBa** [comment]
I'm fine with this being only part of the current package. Still, I would find it most useful if the csv writing is done **by default** whenever the function to save the plot is called.
